### DISCUSSION TITLE
Convert three different csv into a graph and do classification?

### DISCUSSION THREAD

**Discussion by monk1337** [pyg-team/pytorch_geometric#5583] [Q&A]
Hi, I have three different csv files, for example data

<img width="1103" alt="Screenshot 2022-09-30 at 6 09 10 PM" src="https://user-images.githubusercontent.com/17107749/193271779-817cfcb8-1ad1-4be7-8145-57728c9f0c33.png">

I want to use the `qualified` column as a a label and predict based on the variables given in data_dm and data_cm csv files. How to convert these three CSV into graph along with labels and do classification? 

Thank you!

**Comment by rusty1s** [comment]
It doesn't look like your data naturally converts to a graph since tables are only connected based on a 1:1 relationship. Instead, you can flatten these tables together based on `payer_id` and perform logistic regression/XGBoost/MLPs to make predictions per row.

**Reply by monk1337** [reply]
Thank you quick reply. This is demo data, and the actual data contains many columns and hierarchal medical records. Could you share any guidance on how I should proceed to convert such data into a knowledge graph and make node predictions?
My goal is to convert data into a knowledge graph and do node/link classification. But I couldn't find any example of how to convert such data along with labels?

Thank you!

**Reply by rusty1s** [reply]
You may find [this](https://pytorch-geometric.readthedocs.io/en/latest/notes/load_csv.html) tutorial helpful. Let me know.

**Reply by monk1337** [reply]
Thank you for the tutorial; I have already gone through the tutorial, and I have two questions:

1) How to load labels in that tutorial and do node classification?
2) How to use the same tutorial if data has many columns and relationships?

**Reply by rusty1s** [reply]
1. Instead of adding all features to the features, you can select the label column and add it as an separate attribute to your `data`, e.g., `data[node_type].y = torch.from_numpy(df["my_label_column"].values)`.
2. The tutorial should be agnostic to the number of tables/columns, i.e., you call `load_edge_csv` for every PK/FK relationship.
